### ISSUE TITLE
Hot Android phone.

### ISSUE THREAD

**Issue by Ginner** [netbirdio/netbird#1140]
**Describe the problem**
While connected, the Netbird Android app uses very large amounts of power, so much so that my phone gets hot to the touch! 

Further more, the connection is dropped, and when opening the app it says 'Reconnecting...' which it won't unless you tap to disconnect and then connects again (maybe a separate issue?). I've activated 'Always-on VPN' in the settings.

**To Reproduce**
1. Download the apk from the release page and install it.
2. Establish a connection to your network.
3. Enable 'Allways-on VPN' in the settings.
4. Feel the burn.

**Expected behavior**
I'd hope for very little impact on battery life and no noticeable raise in temperature.

Once 'Allways-on VPN' is set, it should stay connected.

**Additional context**
Phone: OnePlus 7Pro
OS: LineageOS, Android 13
Netbird free tier hosted solution. 

Thank you for your awesome work!

**Comment by pappz** [comment]
First of all sorry for the inconvenience. It is sounds strange. The app should not consume huge amount of battery power in normal case. Maybe happen an exception internally and the program try to reconnect in incorrect way. 
Do you have experience with the ADB tool? This debug tool provide an option to collect logs from the Android device. If yes could you send to me the output of the "adb logcat -v time"?

**Comment by Ginner** [comment]
Please don't apologize! I really appreciate all of the work you guys are doing and giving to the community! :)

So I tried hooking up adb and it worked fine, no issues, no heat, stays connected. 

Then I though, it might be because the phone was plugged in, and it could be related to some sort of power saving feature 'turning off' Netbird (although it was set to always-on). 
I unplugged it, and kept my eyes (and hands) on it - no issues, no heat. Then I tried turning wifi off, which coursed the issue to [return](https://ibb.co/Bz5B8Vn). It wouldn't automatically reconnect over 4g, it just said 'Reconnecting...' and was getting more and more hot. As stated before, I could then tap to disconnect, and tap again to connect and it would connect almost instantly. However, the heat/battery drain seemed to persist through this reconnect, until I disconnected entirely. I tried to connect back to wifi, same issue, it'll say 'Reconnecting' indefinitely. 

I've retried disconnecting and connecting while having it connected through adb. Same issue with reconnecting and heat. This time though, I noticed loosing internet connectivity (This happened before, It just quickly returned, as I'd either connect/disconnect netbird manually) and it didn't return, until I manually stopped Netbird's attempt to reconnect. When it happened, it ended the adb stream as well with `09-14 21:10:52.750 I/android.hardware.power-service-qti( 1040): Power setMode: 5 to: 1`. I restarted adb, whilst not having internet, and it returned no problem. 

However, there is a lot(~80.000 over a span of 10-15min!) of lines and I must say, I'm not fully confident in the individual apps sanitizing their logging sufficiently, hence I'm not fully confident in sharing the full log here. Is there a certain app or service output or other relevant info that you'd find interesting, that I could maybe grep/regex out? Or something else I can do?

That got long winded, I'm sorry. And again, thank you for all your work!

**Comment by nazarewk** [comment]
I'm pretty sure this is related to high battery/data usage topics and caused by the amount of computation/retries done on the client side.

Just to let you know we are working on a "lazy connections" feature which should positively affect both data and battery drain issues.

https://github.com/netbirdio/netbird/pull/3379

**Comment by ALParsons** [comment]
Enabling lazy connections addressed the drain issue but carries its own side effects for restoring connection to some types of endpoints, and the configuration is only available at the tenant-level, not per client or per network/resource which would be ideal. 

Is there anything that can be done to reduce the retry frequency? Maybe in particular where always-on is enabled for Netbird (i.e. on Android, Always On VPN is enabled for Netbird VPN client) but otherwise this seems like something that should attempt to be resolved at the cause in addition to lazy-connection enhancements.

**Comment by PizzaLovingNerd** [comment]
We are closing this, as it seems to be solved by lazy connections. Cheers!

**Comment by ALParsons** [comment]
@PizzaLovingNerd is there a back-off algorithm in the client for connection retries?

**Comment by PizzaLovingNerd** [comment]
Yes, the client uses exponential backoff, though peer connections have separate retry logic. We're closing this based on the reported improvement with lazy connections, rather than a confirmed fix to the retry behavior.
